src/components/DuplicateCourseModal.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { useQuery } from 'react-query';
+
+import { Button, Checkbox, Modal, Select } from '@folio/stripes/components';
+import { useOkapiKy } from '@folio/stripes/core';
+
+
+import manifest from '../util/manifest';
+
+const DuplicateCourseModal = ({ data, history, onClose }) => {
+  const intl = useIntl();
+  const ky = useOkapiKy();
+
+  const [term, setTerm] = useState();
+  const { data: terms } = useQuery(
+    ['ui-courses', 'terms'],
+    () => ky(manifest.terms.path).json(),
+    {
+      placeholderData: [],
+      onSuccess: json => setTerm(json[0]?.value),
+      select: json => ((json?.terms ?? [])
+        .map(t => ({ value: t.id, label: t.name }))
+        .sort((a, b) => a.label.localeCompare(b.label))
+      )
+    }
+  );
+
+  const [duplicateCrosslistedCourses, setDuplicateCrosslistedCourses] = useState(false);
+  useQuery(
+    ['ui-courses', 'duplicateCrosslistedCourses'],
+    () => ky(manifest.displaySettings.path).json(),
+    {
+      onSuccess: json => setDuplicateCrosslistedCourses(json.duplicateCrosslistedCourses),
+      select: json => JSON.parse(json.configs[0].value),
+    },
+  );
+
+  const duplicateCourse = async () => {
+    const { course, crossListed, reserves } = data;
+    const { courseListingObject: courseListing } = course;
+
+    const newCourseListing = await ky.post('coursereserves/courselistings', { json: {
+      courseTypeId: courseListing.courseTypeId,
+      externalId: courseListing.externalId,
+      locationId: courseListing.locationId,
+      registrarId: courseListing.registrarId,
+      servicepointId: courseListing.servicepointId,
+      termId: term,
+    } }).json();
+
+    for (let i = 0; i < courseListing.instructorObjects.length; i++) {
+      await ky.post(`coursereserves/courselistings/${newCourseListing.id}/instructors`, { json: {
+        courseListingId: newCourseListing.id,
+        name: courseListing.instructorObjects[i].name,
+        userId: courseListing.instructorObjects[i].userId,
+      } }).json();
+    }
+
+    for (let i = 0; i < reserves.length; i++) {
+      await ky.post(`coursereserves/courselistings/${newCourseListing.id}/reserves`, { json: {
+        courseListingId: newCourseListing.id,
+        itemId: reserves[i].itemId,
+      } }).json();
+    }
+
+    const newCourse = await ky.post(`coursereserves/courselistings/${newCourseListing.id}/courses`, { json: {
+      courseListingId: newCourseListing.id,
+      courseNumber: course.courseNumber,
+      departmentId: course.departmentId,
+      description: course.description,
+      name: `${course.name} - ${intl.formatMessage({ id: 'ui-courses.button.duplicate' })}`,
```

Comment:
I wonder whether re-using the translation tag for the button here is a bit cheeky.

Reply:
More thoughts on this: it works in English because the verb used on the button and the adjectival noun used in the new course's name are both spelled "duplicate"; but it's easy to imagine that in other languages the verb and the noun are different words.

Reply:
Very true. I try to be good about this but completely missed that.